spec/pane-container-element-spec.coffee

```diff
@@ -0,0 +1,110 @@
+
+      expectResizeElement = (index) ->
+        child = paneAxisElement.children[index]
+        expect(paneAxisElement.isPaneResizeHandleElement(child)).toBe(true)
+
+      models = (new PaneAxis({}) for i in [0..2])
+      paneAxis.addChild(models[0])
+      paneAxis.addChild(models[1])
+      expectResizeElement(1)
+
+      paneAxis.addChild(models[2])
+      expectPaneAxisElement(i) for i in [0, 2, 4]
+      expectResizeElement(i) for i in [1, 3]
+
+      # test removeChild
+      paneAxis.removeChild(models[2])
+      expectResizeElement(i) for i in [1]
+
+  describe "when user drag the resize element", ->
+    [container, containerElement] = []
+    [resizeElementMove, getElementWidth, expectPaneScale] = []
+
+    beforeEach ->
+      container = new PaneContainer
+      containerElement = atom.views.getView(container);
+      document.querySelector('#jasmine-content').appendChild(containerElement)
+
+      resizeElementMove = (resizeElement, clientX, clientY) ->
+        mouseDownEvent = new MouseEvent 'mousedown',
+          { view: window, bubbles: true, button: 0 }
+        resizeElement.dispatchEvent(mouseDownEvent)
+
+        mouseMoveEvent = new MouseEvent 'mousemove',
+          { view: window, bubbles: true, clientX: clientX, clientY: clientY}
+        resizeElement.dispatchEvent(mouseMoveEvent)
+
+        mouseUpEvent = new MouseEvent 'mouseup',
+          {view: window, bubbles: true, button: 0}
+        resizeElement.dispatchEvent(mouseUpEvent)
+
+      getElementWidth = (element) ->
+        element.getBoundingClientRect().width
+
+      # assert the pane's flex scale. arguments is list of pane-scale pair
+      expectPaneScale = ->
+        args = Array::slice.call(arguments, 0)
+        for paneScale in args
+          expect(paneScale[0].getFlexScale()).toBeCloseTo(paneScale[1], 0.1)
+
+    it "add and remove panes in the same direction", ->
+      leftPane = container.getActivePane()
+      middlePane = leftPane.splitRight()
+
+      [resizeElements, paneElements] = []
+      reloadElements = ->
+        resizeElements = containerElement.querySelectorAll('atom-pane-resize-handle')
+        paneElements = containerElement.querySelectorAll('atom-pane')
+      reloadElements()
+      expect(resizeElements.length).toBe(1)
+
+      resizeElementMove(resizeElements[0], getElementWidth(paneElements[0]) / 2)
+      expectPaneScale [leftPane, 0.5], [middlePane, 1.5]
+
+      # add a new pane
+      rightPane = middlePane.splitRight()
+      reloadElements()
+      clientX = getElementWidth(paneElements[0]) + getElementWidth(paneElements[1]) / 2
+      resizeElementMove(resizeElements[1], clientX)
+      expectPaneScale [leftPane, 0.5], [middlePane, 0.75], [rightPane, 1.75]
+
+      middlePane.close()
+      expectPaneScale [leftPane, 0.44], [rightPane, 1.55]
+
+      leftPane.close()
+      expectPaneScale [rightPane, 1]
+
+    it "split or close panes in orthogonal direction", ->
```

Comment:
I just want to test that the scale of horizontal panes is not changed when the vertical panes are added or removed.